broadlink/climate.py

```diff
@@ -238,3 +238,393 @@ def set_schedule(self, weekday: List[dict], weekend: List[dict]) -> None:
+        """Returns dictionary with A/C info.
+
+        Args:
+            payload_debug (Optional[bool]): print the received payload
+
+        Returns:
+            dict:
+                state (bool): power
+                ambient_temp (float): ambient temperature
+        """
+        payload = self._send_short_payload(0)
+        if (len(payload) != 48):
+            raise ValueError(f'get_ac_info, unexpected payload size: {len(payload)}')
+
+        # The first 13 bytes are the same: 22 00 bb 00 07 00 00 00 18 00 01 21 c0,
+        # bytes 0x23,0x24 are the checksum
+        # bytes 0x25 forward are always empty
+        data = {}
+        data['state'] = payload[0x0d] & 0b1 == 0b1
+
+        ambient_temp = payload[0x11] & 0b00011111
+        if ambient_temp:
+            data['ambient_temp'] = ambient_temp + float(payload[0x21] & 0b00011111) / 10.0
+
+        checksum = self._calculate_checksum(payload[2:0x23], 'big')
+        if (payload[0x23:0x25] != checksum):
+            print(f'in get_ac_state, checksum fail: calculated '
+                  f'{checksum.hex()} actual {payload[0x23:0x25].hex()}')
+
+        if (payload_debug):
+            print(payload.hex(' '))
+
+        return data
+
+    def set_advanced(
+        self,
+        state: bool,
+        mode: str,
+        target_temp: float,
+        speed: str,
+        swing_v: str,
+        swing_h: str,
+        sleep: bool,
+        display: bool,
+        health: bool,
+        clean: bool,
+        mildew: bool,
+        cmnd_0d_rmask: int = 0b100,
+        cmnd_0e_rmask: int = 0b1101,
+        cmnd_18: int = 0b101,
+        payload_debug: bool = False
+    ) -> bytes:
+        """Set parameters of unit.
+
+        Use `set_partial` to modify only some parameters.
+
+        Args:
+            state (bool): power
+            target_temp (float): temperature set point 16<n<32
+            mode (str): cooling, heating, fan, dry, auto
+            speed (str): mute, low, mid, high, turbo (available only for cooling)
+            swing_h (str): ON, OFF
+            swing_v (str): ON, OFF, 1, 2, 3, 4, 5 (fixed positions)
+            sleep (bool)
+            display (bool)
+            health (bool)
+            clean (bool)
+            mildew (bool)
+            cmnd_0d_rmask (Optional[int]): override an unidentified option
+            cmnd_0e_rmask (Optional[int]): override an unidentified option
+            cmnd_18 (Optional[int]): override an unidentified option
+            payload_debug (Optional[bool]): print the constructed payload
+
+        Returns:
+            True for verified success.
+        """
+
+        PREFIX = [0x19, 0x00, 0xbb, 0x00, 0x06, 0x80, 0x00, 0x00, 0x0f, 0x00,
+                  0x01, 0x01]  # 12B
+        MIDDLE = [0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+                  0, 0, 0, 0, 0]  # 13B + 2B checksum
+        SUFFIX = [0, 0, 0, 0, 0]  # 5B
+        payload = PREFIX + MIDDLE + SUFFIX
```

Comment:
We could have a self._encode() for this, reference [here](https://github.com/mjg59/python-broadlink/blob/29345a129ff6c98962fccba8ada1ea049fd9ec16/broadlink/switch.py#L319-L330).